src/math/geometry.cpp

```diff
@@ -243,6 +243,81 @@ int containedInTriangle(
   return NOT_CONTAINED;
 }
 
+ConvexityResult isConvexQuad(std::array<Eigen::VectorXd, 4> coords)
+{
+  /*
+    All points need to be projected into a new plane with only 2 coordinates, x' and y'. These are used to check 
+    the convexity of the quad. These new coordinates are stored in 'coords'. Vertices in 2D does not need to
+    be transformed.
```

Comment:
```suggestion
    All points need to be projected into a new plane with only 2 coordinates, x' and y'. These are used to check 
    the convexity of the quad. These new coordinates are stored in 'coords'.
```
2D makes no sense here. Could you please add an assertion to safeguard against that?